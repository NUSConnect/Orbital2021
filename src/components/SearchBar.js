import React from "react";
import {
    StyleSheet,
    TextInput,
    View,
    Image,
    TouchableOpacity,
} from "react-native";

const SearchBar = ({search, setSearch, searchFilterFunction, resetFilter}) => {

    return (
        <View style={styles.searchBar}>
            <TextInput
                style={styles.textInputStyle}
                onChangeText={(text) => searchFilterFunction(text)}
                value={search}
                placeholder="Search Here"
            />
            {search !== "" ? (
                <TouchableOpacity
                    style={styles.closeButtonParent}
                    onPress={() => {
                        resetFilter();
                        setSearch("");
                    }}
                >
                    <Image
                        style={styles.closeButton}
                        source={require("../assets/close-circle-outline.png")}
                    />
                </TouchableOpacity>
            ) : null}
        </View>
    )
}

export default SearchBar;

const styles = StyleSheet.create({
    textInputStyle: {
        height: 40,
        paddingLeft: 20,
        flex: 1,
    },
    searchBar: {
        flexDirection: "row",
        margin: 5,
        borderColor: "#ff8c00",
        backgroundColor: "#FFFFFF",
        borderWidth: 1,
    },
    closeButtonParent: {
        justifyContent: "center",
        alignItems: "center",
        marginRight: 10,
    },
    closeButton: {
        height: 20,
        width: 20,
    },
});