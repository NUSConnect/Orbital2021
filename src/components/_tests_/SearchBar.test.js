import * as React from 'react'
import SearchBar from '../SearchBar'
import renderer from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react-native'

jest.useFakeTimers()
jest.mock('react-native-vector-icons')

it('renders correctly', () => {
  const tree = renderer.create(<SearchBar />)
  expect(tree).toMatchSnapshot()
})

it('onChangeText works', () => {
  const setSearchFn = jest.fn()
  const filterFn = jest.fn()
  const resetFilterFn = jest.fn()
  const { getByTestId } = render(
    <SearchBar search='a' setSearch={setSearchFn} searchFilterFunction={filterFn} resetFilter={resetFilterFn} />
  )

  fireEvent.changeText(getByTestId('input'), 'test')
  expect(filterFn).toHaveBeenCalledWith('test')
})

it('reset button works', () => {
  const setSearchFn = jest.fn()
  const filterFn = jest.fn()
  const resetFilterFn = jest.fn()
  const { getByTestId } = render(
    <SearchBar search='a' setSearch={setSearchFn} searchFilterFunction={filterFn} resetFilter={resetFilterFn} />
  )

  fireEvent.press(getByTestId('reset'))
  expect(setSearchFn).toHaveBeenCalledWith('')
  expect(resetFilterFn).toHaveBeenCalled()
})
