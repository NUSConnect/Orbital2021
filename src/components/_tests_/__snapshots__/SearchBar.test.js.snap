// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#ff8c00",
      "borderWidth": 1,
      "flexDirection": "row",
      "margin": 5,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Search Here"
    rejectResponderTermination={true}
    style={
      Object {
        "flex": 1,
        "height": 40,
        "paddingLeft": 20,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginRight": 10,
        "opacity": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
</View>
`;
